scripts/generate_async_api.py

```diff
@@ -40,15 +40,16 @@ def generate(t: Any) -> None:
     class_name = short_name(t)
     base_class = t.__bases__[0].__name__
     if class_name in ["Page", "BrowserContext", "Browser"]:
-        base_sync_class = "AsyncContextManager"
+        base_async_class = "AsyncContextManager"
     elif base_class in ["ChannelOwner", "object"]:
-        base_sync_class = "AsyncBase"
+        base_async_class = "AsyncBase"
     else:
-        base_sync_class = base_class
-    print(f"class {class_name}({base_sync_class}):")
+        base_async_class = base_class
+    print("@final")
+    print(
+        f"class {class_name}({base_async_class}): {'# type: ignore' if base_async_class== 'JSHandle' else ''}"
```

Comment:
`# type: ignore` is for `ElementHandle` as it inherits from `JSHandle`

Reply:
is the `@final` thing only for linting or does it actually throw when you try to extend from it again?

Reply:
don't think its `runtime_checkable`, if it's only for mypy this is fine, _enforce_ it through other means (__init_subclass__? or metaclass I guess) but not sure if that's what you want to enforce